src/services/AppUserService.ts

```diff
@@ -18,4 +32,34 @@ export class AppUserService {
   getMultipleAppUsers(ids: number[]): Promise<AppUser[]> {
     return this.appUserRepository.find({ id: Any(ids) });
   }
+
+  /**
+   * Get an existing AppUser by their email address.
+   *
+   * @param email The email used to look for the corresponding AppUser.
+   *
+   */
+  getAppUserByEmail(email: string): Promise<AppUser> {
+    return this.appUserRepository.findOne({ email });
+  }
+
+  async saveNonAffiliate(appUserRequest: EventSignInRequest): Promise<AppUser | undefined> {
```

Comment:
thoughts on doing the mapping in the controller instead? that way services dont have to know about payloads :)